File: library/src/main/java/org/incendo/interfaces/inventory/PlayerInterfacesInventory.java
package org.incendo.interfaces.inventory;

import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.incendo.interfaces.utilities.BukkitInventoryUtilities;

public final class PlayerInterfacesInventory extends CachedInterfacesInventory {

    private final Player player;

    public PlayerInterfacesInventory(final Player player) {
        this.player = player;
    }

    @Override
    public ItemStack get(final int row, final int column) {
        int index = BukkitInventoryUtilities.gridPointToBukkitIndex(row, column);
        return this.player.getInventory().getItem(index);
    }

    @Override
    public void setInternal(final int row, final int column, final ItemStack item) {
        int index = BukkitInventoryUtilities.gridPointToBukkitIndex(row, column);
        this.player.getInventory().setItem(index, item);
    }

    @Override
    public boolean isPlayerInventory(final int row, final int column) {
        return true;
    }
}

File: next/src/main/kotlin/org/incendo/interfaces/next/inventory/PlayerInterfacesInventory.kt
package org.incendo.interfaces.next.inventory

import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.incendo.interfaces.next.utilities.gridPointToBukkitIndex

public class PlayerInterfacesInventory(
    private val player: Player,
) : CachedInterfacesInventory() {
    private val playerInventory = player.inventory

    override fun get(
        row: Int,
        column: Int,
    ): ItemStack? {
        val index = gridPointToBukkitIndex(row, column)
        return playerInventory.getItem(index)
    }

    override fun setInternal(
        row: Int,
        column: Int,
        item: ItemStack?,
    ) {
        val index = gridPointToBukkitIndex(row, column)
        return playerInventory.setItem(index, item)
    }

    override fun isPlayerInventory(
        row: Int,
        column: Int,
    ): Boolean = true
}
